Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -444,31 +354,138 @@ namespace AzFramework
         TargetManagerClient::Bus::ExecuteQueuedEvents();
     }
 
-    void TargetManagementComponent::JoinNeighborhood()
+    bool TargetManagementComponent::HandleRequest(
+        [[maybe_unused]] AzNetworking::IConnection* connection,
+        [[maybe_unused]] const AzNetworking::IPacketHeader& packetHeader,
+        [[maybe_unused]] const AzFrameworkPackets::TargetConnect& packet)
     {
-        // if we don't have a specified neighborhood name, don't try to connect.
-        if (m_settings->m_neighborhoodName.empty())
+        EBUS_EVENT(Neighborhood::NeighborhoodBus, OnNodeJoined, packet, static_cast<uint32_t>(connection->GetConnectionId()));
+        return true;
+    }
+
+    bool TargetManagementComponent::HandleRequest(
+        [[maybe_unused]] AzNetworking::IConnection* connection,
+        [[maybe_unused]] const AzNetworking::IPacketHeader& packetHeader,
+        [[maybe_unused]] const AzFrameworkPackets::TargetMessage& packet)
+    {
+        // Receive
+        if (connection->GetConnectionRole() == AzNetworking::ConnectionRole::Acceptor &&
+            static_cast<uint32_t>(connection->GetConnectionId()) != m_settings->m_lastTarget.m_networkId)
         {
-            return;
+            // Listener should route traffic based on selected target
+            return true;
+        }
+
+        // If we're a client, set the host to our desired target
+        if (connection->GetConnectionRole() == AzNetworking::ConnectionRole::Connector)
+        {
+            if (GetTargetInfo(packet.GetPersistentId()).GetPersistentId() == 0)
+            {
+                TargetContainer::pair_iter_bool ret = m_availableTargets.insert_key(packet.GetPersistentId());
+
+                TargetInfo& ti = ret.first->second;
+                ti.m_lastSeen = time(nullptr);
+                ti.m_displayName = "Host";
+                ti.m_persistentId = packet.GetPersistentId();
+                ti.m_networkId = static_cast<uint32_t>(connection->GetConnectionId());
+                ti.m_flags |= TF_ONLINE | TF_DEBUGGABLE | TF_RUNNABLE;
+
+                // last target can be loaded in with a persistent id but no network, correct here
+                if (GetDesiredTarget().GetPersistentId() == packet.GetPersistentId())
+                {
+                    m_settings->m_lastTarget.m_networkId = ti.m_networkId;
+                }
+
+                EBUS_QUEUE_EVENT(TargetManagerClient::Bus, TargetJoinedNetwork, ti);
+            }
+
+            if (GetDesiredTarget().GetPersistentId() != packet.GetPersistentId())
+            {
+                SetDesiredTarget(packet.GetPersistentId());
+            }
+        }
+
+        const uint32_t totalBufferSize = packet.GetSize();
+
+        if (m_tmpInboundBufferPos == 0)
+        {
+            m_tmpInboundBuffer.reserve(totalBufferSize);
+        }
+
+        const uint32_t readSize = AZStd::min(totalBufferSize - m_tmpInboundBufferPos, Neighborhood::NeighborBufferSize);
```

Comment:
nit: Consider adding comment here to cover the expectations below, its some dense code to get through so comment would help parse expected behaviour